File: 1/1.hs
import System.IO
import Data.Function

isGreater :: Integer -> Integer -> Integer
isGreater x y
  | y > x     = 1
  | otherwise = 0

countIncreases :: [Integer] -> Integer
countIncreases x = sum $ zipWith isGreater x $ tail x

slidingWindowSum :: Integer -> [Integer] -> [Integer]
slidingWindowSum windowSize list
  | windowSize == 1 = list
  | otherwise       = zipWith (+) list $ slidingWindowSum (windowSize - 1) $ tail list

main :: IO()
main = do
    inputs <- readFile "input.txt"
    let xs = map (\x -> read x::Integer) (lines inputs)
    print $ countIncreases xs
    print $ countIncreases $ slidingWindowSum 3 xs

File: 1/1.py
## STAR 1
def star1():
    with open("input.txt", "r") as f:
        data = [int(x) for x in f.readlines()]
        return (sum(x < y for x, y in zip(data, data[1:])))

## STAR 2
def star2():
    with open("input.txt", "r") as f:
        data = [int(x) for x in f.readlines()]
        grouped = [sum(data[i:i + 3]) for i in range(len(data) - 2)]
        return (sum(x < y for x, y in zip(grouped, grouped[1:])))

def main():
    print("Star 1: ", star1())
    print("Star 2: ", star2())

if __name__ == '__main__':
    main()